modules/packages/OrderedSet/Treap.chpl

```diff
@@ -0,0 +1,872 @@
+/*
+ * Copyright 2004-2020 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  This module contains a implementation of Treap,
+  which provides the functionality of OrderedSet.
+  Treap supporst insertion, deletion, query in O(lgN).
+
+  .. note::
+    Generally, users don't have to directly use this module. The methods of a treap 
+    are avaliable for an orderedSet. This page is for reference.
+*/
+module Treap {
+  import ChapelLocks;
```

Comment:
There's only one `private import` in all of `modules/`.
(And `use` defaults to private everywhere, except `modules/internal` for now, right?)

Reply:
That's true. I just thought it would be clearer if we added it here. I am also OK with dropping other `private`s from `use`s. But I personally think we should be clear in import/uses in our modules.

Reply:
I more feel like to follow other modules and use `import`.

Reply:
OK